Tests/Editor/CoreApi/ManagedEC2DeploymentTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using AmazonGameLift.Editor;
+using AmazonGameLiftPlugin.Core.SettingsManagement.Models;
+using AmazonGameLiftPlugin.Core.Shared;
+using Editor.CoreAPI;
+using Moq;
+using NUnit.Framework;
+using UnityEngine;
+
+namespace AmazonGameLiftPlugin.Editor.UnitTests
+{
+    public class ManagedEC2DeploymentTests
+    {
+        private AwsCredentialsTestProvider _awsCredentialsTestProvider;
+        private Mock<ManagedEC2FleetParameters> _managedEc2FleetParametersMock;
+        private DeploymentSettings _deploymentSettings;
+        private Mock<ConfirmChangesRequest> _confirmChangesRequestMock;
+        private static Mock<CoreApi> _coreApiMock;
+        
+        [SetUp]
+        public void Setup()
+        {
+            _awsCredentialsTestProvider = new AwsCredentialsTestProvider();
+            _managedEc2FleetParametersMock = new Mock<ManagedEC2FleetParameters>();
+            //_confirmChangesRequestMock = new Mock<ConfirmChangesRequest>();
+            _coreApiMock = new Mock<CoreApi>();
+        }
+
+        private ManagedEC2Deployment ArrangeEc2DeploymentHappyPath()
+        {
+            _coreApiMock.Setup(f => f.PutSetting(It.IsAny<SettingsKeys>(), It.IsAny<string>())).Returns(Response.Ok(new PutSettingResponse()));
+            _coreApiMock.Setup(f => f.PutSettingOrClear(It.IsAny<SettingsKeys>(), It.IsAny<string>())).Returns(Response.Ok(new PutSettingResponse()));
+            _coreApiMock.Setup(f => f.PutSetting(It.IsAny<SettingsKeys>(), null)).Returns(Response.Fail(new PutSettingResponse()));
+            _coreApiMock.Setup(f => f.PutSetting(It.IsAny<SettingsKeys>(), string.Empty)).Returns(Response.Fail(new PutSettingResponse()));
+            _coreApiMock.Setup(f => f.GetSetting(It.IsAny<SettingsKeys>())).Returns(Response.Ok(new GetSettingResponse()));
+            
+            _deploymentSettings = GetUnitUnderTest();
+            return new ManagedEC2Deployment(_deploymentSettings, _managedEc2FleetParametersMock.Object);
+        }
+        
+        [Test]
+        public void StartDeployment_WhenCorrectInputs_ExpectSuccess()
+        {
+            //Arrange
+            var ec2DeploymentHappyPath = ArrangeEc2DeploymentHappyPath();
+            
+            //Act
+            ec2DeploymentHappyPath.StartDeployment();
+            
+            //Assert
+            _coreApiMock.Verify(f => f.PutSetting(It.IsAny<SettingsKeys>(), It.IsAny<string>()), Times.Exactly(8));
+            
+            Assert.IsTrue(_deploymentSettings.GameName == Application.productName.Substring(0,12) );
+        }
+        
+        [Test]
+        public void StartDeployment_WhenTaskIsFaulted_ExpectException()
```

Comment:
ExpectException implies there's an exception we can verify. We aren't doing that in this test

Reply:
Changed to actually expect Exception